pages/visualizations/overview/issues_over_time.py

```diff
@@ -127,96 +127,125 @@ def issues_over_time_graph(repolist, timer_pings, interval):
     start = time.perf_counter()
 
     # create dataframe from record data
-    df_issues = pd.DataFrame(results)
+    df = pd.DataFrame(results)
+
+    # test if there is data
+    if df.empty:
+        logging.debug("ISSUES OVER TIME - NO DATA AVAILABLE")
+        return nodata_graph, False, dash.no_update
+
+    # convert to datetime objects rather than strings
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+    df["closed"] = pd.to_datetime(df["closed"], utc=True)
 
     # order values chronologically by creation date
-    df_issues = df_issues.sort_values(by="created")
+    df = df.sort_values(by="created", axis=0, ascending=True)
+
+    # variable to slice on to handle weekly period edge case
+    period_slice = None
+    if interval == "W":
+        period_slice = 10
+
+    # data frames for issues created, merged, or closed. Detailed description applies for all 3.
+
+    # get the count of created issues in the desired interval in pandas period format, sort index to order entries
+    created_range = pd.to_datetime(df["created"]).dt.to_period(interval).value_counts().sort_index()
+
+    # converts to data frame object and creates date column from period values
+    df_created = created_range.to_frame().reset_index().rename(columns={"index": "Date"})
+
+    # converts date column to a datetime object, converts to string first to handle period information
+    # the period slice is to handle weekly corner case
+    df_created["Date"] = pd.to_datetime(df_created["Date"].astype(str).str[:period_slice])
+
+    # df for merged issues in time interval
+    closed_range = pd.to_datetime(df["closed"]).dt.to_period(interval).value_counts().sort_index()
+    df_closed = closed_range.to_frame().reset_index().rename(columns={"index": "Date"})
+    df_closed["Date"] = pd.to_datetime(df_closed["Date"].astype(str).str[:period_slice])
+
+    if interval == "M":
+        df_created["Date"] = df_created["Date"].dt.strftime("%Y-%m-01")
+        df_closed["Date"] = df_closed["Date"].dt.strftime("%Y-%m-01")
+    elif interval == "Y":
+        df_created["Date"] = df_created["Date"].dt.strftime("%Y-01-01")
+        df_closed["Date"] = df_closed["Date"].dt.strftime("%Y-01-01")
+
+    # first and last elements of the dataframe are the
+    # earliest and latest events respectively
+    earliest, latest = df.iloc[0]["created"], df.iloc[-1]["created"]
```

Comment:
the earliest is definitely a 'created' event but the latest is the most recent 'created' or 'closed' event, so you'd take the max of (df.iloc[-1]['created'], df.iloc[-1]['closed]) to get the most recent.